### DISCUSSION TITLE
A few small questions

### DISCUSSION THREAD

**Discussion by CaiB** [thuliteio/doks#64] [Q&A] [answered]
I'm putting the finishing touches on my website, thanks a lot for this awesome theme. I had a few small lingering questions:

By default, the use of screen space on the documentation pages is very wasteful on desktop-size displays. I decided to try and fix this by increasing the max width as such:
`assets/scss/common/_veriables.scss:48`:
```SCSS
$container-max-widths: (
  sm: 540px,
  md: 720px,
  lg: 960px,
  xl: 90%
);
```
Changing `xl` to a large percentage of the screen makes the content fill up much more of the horizontal space. It appears to work well over a range of window sizes, but I wanted to double-check that this wasn't going to have some unintended bad consequences. I am not a web developer, so this seems fine to me, but please let me know if I'm being dumb.

---

Is there a way to default to using the dark theme on first visit, or perhaps even an easy way to disable the light theme entirely?

**Comment by h-enk** [comment, marked as answer]
Nice to hear that you like Doks! I'm curious what you're building 😉 

## xl container

The [xl container](https://getbootstrap.com/docs/4.5/layout/grid/#grid-options) is for screens with a viewport of ≥1200px. Setting the xl container to a relative width (90% i.s.o. 1140px — like you did) is OK /works alright. Then, only the main content on a documentation page grows/shrinks and the sidebars remain fixed. Nice customization.

## Dark mode

The dark/light toggle sets/removes class `dark` to/from the `body` tag. 

### Start with dark mode

If you want to start with dark mode, add `dark` to the body class in `./layouts/_default/baseof.html` (line 15):

```html
<body class="{{ .Scratch.Get "class" }} dark">
```

### Disable light theme

If you want to disable the light theme:

1. Add `dark` to the body class (see above: Start with dark mode).
2. Comment out lines 1..12 of `./assets/js/app.js`:

```js
/*
document.getElementById('mode').addEventListener('click', () => {

  document.body.classList.toggle('dark');
  localStorage.setItem('theme', document.body.classList.contains('dark') ? 'dark' : 'light');

});

if (localStorage.getItem('theme') === 'dark') {

  document.body.classList.add('dark');

}
*/
```
3. Comment out lines 7..10 of `./layouts/partials/header/header.html`:
```html
<!--
<button id="mode" class="btn btn-link order-2 order-md-4" type="button" aria-label="Toggle mode">
  <span class="toggle-dark"><svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-moon"><path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path></svg></span>
  <span class="toggle-light"><svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-sun"><circle cx="12" cy="12" r="5"></circle><line x1="12" y1="1" x2="12" y2="3"></line><line x1="12" y1="21" x2="12" y2="23"></line><line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line><line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line><line x1="1" y1="12" x2="3" y2="12"></line><line x1="21" y1="12" x2="23" y2="12"></line><line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line><line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line></svg></span>
</button>
-->
```

**Reply by CaiB** [reply]
Thank you for the very detailed response! Got it working how I wanted.

I'm making a website for one of my projects here: https://www.colorchord.net/  
Having all the documentation on the GitHub readme was getting a little difficult to navigate, so I thought it would be nice to have a better organized place to put it :)

Thanks for all your help!

**Reply by h-enk** [reply]
You're welcome — nice to see what you've made out of Doks. Good luck with the ColorChord.NET project!

**Reply by mrdotkg** [reply]
Appreciate your time and work. I am using Doks for my new website. NPM with Hugo is fitting nicely into my dev workflow.

I am extending question asked by @CaiB 

How can I configure **To start with Dark mode first and then the toggle to switch between Dark/Light**. I am sharing some observations-

1) Putting `dark class` in the body tag makes all the pages always render with dark theme. (_FINE_)
2) Now as a user, I press toggle theme, to change to light theme. Website is now in light theme. (_FINE_)
3) I reload the page(or browse other pages) and website goes back to dark mode.(_NOT FINE_)

`Doks` remembers theme preference only if we start with light theme first. Not the other way round with your hack @h-enk.

@h-enk Could you also guide me to the source code that toggles dark and light on each page load. I looked into the references you shared but they are merely setting the click event. Where do we trigger this event every time page loads?

@epogrebnyak @h-enk 

Thank you.

**Reply by kevinrpb** [reply]
Hi @mrdotkg! Doks remembers your theme choice by using the LocalStorage API. You can see it in action in two steps:

```javascript
// assets/js/app.js

document.getElementById('mode').addEventListener('click', () => {
  document.body.classList.toggle('dark');

  // 1. Here we save it !
  localStorage.setItem('theme', document.body.classList.contains('dark') ? 'dark' : 'light');
});

// 2. Here we retrieve it !
if (localStorage.getItem('theme') === 'dark') {
  document.body.classList.add('dark');
}
```

As you mentioned, this only works when starting out with the light theme. If you would like it to work the other way around, my solution was replacing step 2 with (this is just an example, you can work it your way just using that LocalStorage value):

```javascript
// 2. Here we retrieve it !
const localTheme = localStorage.getItem('theme');

if (localTheme === 'dark')
  document.body.classList.add('dark');
else if (localTheme === 'light')
  document.body.classList.remove('dark');
```

Just note that if the user never switched the theme, the value in LocalStorage will be null/undefined (that's why I check if it's equal to either theme).

**Reply by h-enk** [reply]
@kevinrpb, nicely done and explained — thanks!

**Reply by mrdotkg** [reply]
Thank you! this works.

In place of `app.js` I have put your code @kevinrpb into `layouts/partials/header/header.html#1` to avoid a noticeable flicker in page loads. As soon as body tag is loaded, the same scripts from `header.html` now will switch Doks theme from dark to light or vice-versa **immediately (theme transition is not noticeable)**.

**Reply by OCram85** [reply]
@h-enk: It really would be nice if there is a *starter* config key  in the `params.toml` like `startTheme = light | dark` to simplify this. So user won't have to customize this part and could easier upgrade doks :D
